packages/fx-core/src/component/driver/teamsApp/createAppPackage.ts

```diff
@@ -111,43 +116,60 @@ export class CreateAppPackageDriver implements StepDriver {
 
     const appDirectory = path.dirname(hasTTKGeneratedFolder ? generatedFolder : manifestPath);
 
-    const colorFile = path.resolve(appDirectory, manifest.icons.color);
-    if (!(await fs.pathExists(colorFile))) {
-      const error = new FileNotFoundError(
-        actionName,
-        colorFile,
-        "https://aka.ms/teamsfx-actions/teamsapp-zipAppPackage"
-      );
-      return err(error);
-    }
-    const colorFileRelativePath = path.relative(appDirectory, colorFile);
-    if (colorFileRelativePath.startsWith("..")) {
-      return err(new InvalidFileOutsideOfTheDirectotryError(colorFile));
-    }
-
-    const outlineFile = path.resolve(appDirectory, manifest.icons.outline);
-    if (!(await fs.pathExists(outlineFile))) {
-      const error = new FileNotFoundError(
-        actionName,
-        outlineFile,
-        "https://aka.ms/teamsfx-actions/teamsapp-zipAppPackage"
-      );
-      return err(error);
+    // check and include all relative file paths in manifest
+    const relativeFiles = [manifest.icons.color, manifest.icons.outline];
+    const manifestVersion = semver.coerce(manifest.manifestVersion); // ensure manifestVersion is a valid semver
+    if (manifestVersion && semver.gte(manifestVersion, "1.21.0")) {
```

Comment:
Do we use the oldest version schema for each of manifest property? These different versions of manifest schema looks hard to maintain.